app.rb

```diff
@@ -0,0 +1,142 @@
+require 'date'
+require_relative 'library/classes/person'
+require_relative 'library/classes/teacher'
+require_relative 'library/classes/student'
+require_relative 'library/classes/book'
+require_relative 'library/classes/rental'
+
+# This class represents the main application for managing books, people, and rentals.
+
+# rubocop:disable Metrics
+class App
+  def display_menu
+    puts 'Please select an option by entering a number'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a Book'
+    puts '5. Create a Rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Exit'
+  end
+
+  def run
+    loop do
+      books_count = Book.books_count
+      Person.people_count
+      Rental.rentals_count
+      students_count = Student.students_count
+      teachers_count = Teacher.teachers_count
+
+      display_menu
+
+      # choice
+      choice = gets.chomp.to_i
+
+      # case choice
+      case choice
+      when 1
+        puts "We have #{books_count} book."
+        if books_count.positive?
+          puts 'Here is the list of all books'
+          Book.list_all_books
+        end
+      when 2
+        puts "[Students] (#{students_count})"
+        Student.list_all_students
+        puts "[Teachers] (#{teachers_count})"
+        Teacher.list_all_teachers
```

Comment:
- [ ] Kindly note that in displaying the list of persons, th key parameters are: `class(teacher or student)`, `name`, `id` and `age`. Yours has only `id` and `name`. Kindly fix the rmaining parameters:

### Video prmpt:
![image](https://github.com/diarisdiakite/oop_school-library/assets/50990925/9f4e280a-a69f-442c-aaba-6a5f92990ea0)

###Your prompt:
![image](https://github.com/diarisdiakite/oop_school-library/assets/50990925/6a33623f-8b4f-429f-accd-8fdd789cc33f)

Reply:
Done @Shedrack-Sunday, Thanks.